src/phylo/node_basics.jl

```diff
@@ -0,0 +1,335 @@
+end
+
+
+"""
+    isclade(tree::Phylogeny, vertex::SymString)
+
+Test whether a node is internal, i.e. has a parent and one or more children.
+
+# Examples:
+```julia
+isclade(my_tree, :Node7)
+isclade(my_tree, "Homo Sapiens")
+isclade(my_tree, Symbol("Homo Sapiens"))
+```
+"""
+function isclade(tree::Phylogeny, name::SymString)
+    return isclade(tree, tree.vertexindex[name])
+end
+
+
+
+"""
+    isleaf(tree::Phylogeny, vertex::Int)
+
+Test whether a vertex is a leaf.
+
+A vertex is a leaf, when it has a parent node, but no children nodes.
+
+# Examples:
+```julia
+isleaf(my_phylogeny, 1)
+```
+"""
+function isleaf(tree::Phylogeny, vertex::Int)
+    return vertex <= tree.ntaxa
+end
+
+
+"""
+    isleaf(tree::Phylogeny, name::SymString)
+
+Test whether a vertex is a leaf.
+
+A vertex is a leaf, when it has a parent node, but no children nodes.
+
+# Examples:
+```julia
+isleaf(my_phylogeny, :A)
+isleaf(my_phylogeny, "A")
+```
+"""
+function isleaf(tree::Phylogeny, name::SymString)
+    return isleaf(tree, tree.vertexindex[name])
+end
+
+
+
+function isclade(tree::Phylogeny, vertex::Int)
+    return vertex > tree.ntaxa
+end
+
+
+
+"""
+    hasparent(tree::Phylogeny, vertex::Int)
+
+Check if a vertex in a phylogeny is linked to a parent.
+"""
+function hasparent(tree::Phylogeny, vertex::Int)
+    return indegree(tree.graph, vertex) == 1
+end
+
+
+"""
+    hasparent(tree::Phylogeny, name::SymString)
+
+Check if the vertex in a phylogeny is linked to a parent.
+"""
+function hasparent(tree::Phylogeny, name::SymString)
+    return hasparent(tree, tree.vertexindex[name])
+end
+
+
+"""
+    hasparent(tree::Phylogeny, vertices::AbstractVector{Int})
+
+Check some vertices in a phylogeny and test if each each is linked to a parent.
+"""
+function hasparent(tree::Phylogeny, vertices::AbstractVector{Int})
+    return indegree(tree.graph, vertices) .== 1
+end
+
+
+
+"""
+    noparent(tree::Phylogeny, vertex::Int)
+
+Check if a vertex in a phylogeny is not linked to a parent.
+"""
+function noparent(tree::Phylogeny, vertex::Int)
+    return indegree(tree.graph) == 0
+end
+
+
+"""
+    noparent(tree::Phylogeny, vertices::AbstractVector{Int})
+
+Check some vertices in a phylogeny and test if each each is not linked to a
+parent.
+"""
+function noparent(tree::Phylogeny, vertices::AbstractVector{Int})
+    return indegree(tree.graph, vertices) .== 0
+end
+
+
+
+"""
+    parent(x::PhyNode, vertex::Int)
+
+Get the parent of a node. In a phylogeny it is not possible
+for any one node/vertex to have more than one parent.
+
+# Examples:
+```julia
+parent(my_tree, 2)
+```
+"""
+function parent(tree::Phylogeny, vertex::Int)
+    @assert hasparent(tree, vertex) error("Vertex does not have a parent.")
+    return in_neighbours(tree.graph, vertex)[1]
+end
+
+
+
+"""
+    is_parent_of(tree::Phylogeny, child::Int, parent::Int)
+
+Test if a vertex of a phylogeny is the parent of another vertex.
+"""
+function is_parent_of(tree::Phylogeny, parent::Int, child::Int)
```

Comment:
Overloading the `hasparent` function may be better (`hasparent(tree, child, parent)`).